future-apps/payroll/test/contracts/Payroll_rates.test.js

```diff
@@ -0,0 +1,142 @@
+const PAYMENT_TYPES = require('../helpers/payment_types')
+const { bigExp, bn } = require('../helpers/numbers')(web3)
+const { getEventArgument } = require('../helpers/events')
+const { deployErc20TokenAndDeposit, deployContracts, createPayrollAndPriceFeed } = require('../helpers/deploy.js')(artifacts, web3)
+
+contract('Payroll rates handling,', ([owner, employee]) => {
+  let USD, ETH, DAI, ANT
+  let dao, payroll, payrollBase, finance, vault, priceFeed, employeeId
+
+  const NOW = 1553703809 // random fixed timestamp in seconds
+  const TWO_MINUTES = 60 * 2
+  const ONE_MONTH = 60 * 60 * 24 * 31
+  const RATE_EXPIRATION_TIME = ONE_MONTH
+
+  const TOKEN_DECIMALS = 18
+  const ONE = bigExp(1, 18)
+  const SIG = '00'.repeat(65) // sig full of 0s
+
+  const formatRate = (n) => bn(n.toFixed(18)).times(ONE)
+
+  const increaseTime = async seconds => {
+    await payroll.mockIncreaseTime(seconds)
+    await priceFeed.mockIncreaseTime(seconds)
+  }
+
+  before('deploy contracts and tokens', async () => {
+    ({ dao, finance, vault, payrollBase } = await deployContracts(owner))
+
+    USD = '0x0000000000000000000000000000000000000001'
+    ETH = '0x0000000000000000000000000000000000000000'
+    await finance.deposit(ETH, bn(50, 18), 'Initial ETH deposit', { from: owner, value: bn(50, 18) })
+
+    DAI = await deployErc20TokenAndDeposit(owner, finance, 'DAI', TOKEN_DECIMALS)
+    ANT = await deployErc20TokenAndDeposit(owner, finance, 'ANT', TOKEN_DECIMALS)
+  })
+
+  beforeEach('initialize payroll app with USD as denomination token', async () => {
```

Comment:
Let's cover our bases and also add some tests for when the Payroll app is initialized with ETH as its denomination token (as the next most probable use case outside of fiat/DAI denomination token).

For this case, let's also use the denomination as the base pricefeed token to test the "non-inverse" case where the org's denomination token matches the pricefeed's base token.